dev/benchmarks/macrobenchmarks/test/util.dart

```diff
@@ -0,0 +1,192 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/scheduler.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:macrobenchmarks/common.dart';
+import 'package:e2e/e2e.dart';
+import 'package:macrobenchmarks/main.dart' as app;
+
+/// The maximum amount of time considered safe to spend for a frame's build
+/// phase. Anything past that is in the danger of missing the frame as 60FPS.
+Duration kBuildBudget = const Duration(milliseconds: 16);
+
+const String _kDebugWarning = '''
```

Comment:
When this PR is out of draft, we should try to minimize duplicate code as it's hard to maintain.

Reply:
If you mean re-use the warning string from `packages/flutter_driver/lib/src/driver/vmservice_driver.dart`, that one is private and need an extra line break to align the warning message.

Reply:
And although it's out of draft, I'm still thinking it as experimental and expecting to see how it will go with device lab. If it's going on well, I think it will worth discussing if `FrameTimingSummarizer` and `watchPerformance` should go to e2e or `flutter_test`.

Reply:
As [`FrameTiming`] is in the SDK, I wonder if we can also put `FrameTimingSummarizer ` in the SDK just in case that developers also want to monitor their app performance in the wild (e.g., https://github.com/flutter/flutter/issues/37437). What do you think, @Hixie ?

[`FrameTiming`]: https://api.flutter.dev/flutter/dart-ui/FrameTiming-class.html

Reply:
`flutter_driver` cannot import things from `macrobenchmarks` because the latter is not a published package and is not a dependency of `flutter_driver`.  There's another similar thing in https://github.com/flutter/flutter/blob/07caa0fbfe5186cc8a11902cc46377cebd592088/packages/flutter_test/lib/src/widget_tester.dart#L324-L334 . I think it might be better to put in `flutter_test`, if we have to.

Reply:
And these three are not exactly the same. One is `flutter run`, one is `flutter drive` (and I just noticed a nit, the one in `vmservice_driver.dart` should have an additional `-t` (#61908) and E2E should be both (just updated this)

Reply:
Done with modification to `flutter_test`